### PR TITLE
FE-1446 - Adds unit tests and CI configuration

### PR DESCRIPTION
**Pull request by nicklemmon**
### What Changed?

- Installs and configures `jest`
- Adds PR template
- Updates Prettier config to match 2web2ui
- Refactors a lot of code for the sake of legibility/modularity
- Adds JSDoc comments to the majority of helper functions
- Adds supporting tests

#### Does it Work?

1. Run Security Badger locally (read README.md for more - if you can't figure out how to run it then the README needs work!)
2. Try and get Security Badger to post to the `#front-end-guild` channel

#### Testing

- Read through tests and verify they adequately test Security Badger

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "typeAcquisition": {
```
Comment (nicklemmon) [inline jsconfig.json:2]: This makes intellisense work for implicit Jest globals (e.g., `it` and `describe`)

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
+/*
```
Comment (nicklemmon) [inline jest.config.js:1]: This config file was generated by Jest

File: README.md [hunk 3]
```diff
@@ -18,11 +20,39 @@ jobs:
   pr-reviews-reminder:
     runs-on: ubuntu-latest
     steps:
-      uses: nicklemmon/security-badger@v0.0.13-alpha
+      uses: SparkPost/security-badger@main
```
Comment (bkemper) [inline README.md:23]: `@main` is needed because it is not a published packaged?

Comment (nicklemmon) [reply README.md:23]: Right - and this way, all projects using Security Badger automatically consume updates. Don't want to have to manually tweak this version across a bunch of different repos for something that isn't terribly important.

File: src/api.js [hunk 4]
```diff
@@ -1,75 +1,74 @@
...
+/**
+ * @description returns structured query data derived from the GitHub repo string
+ * @param {string} githubRepo - GitHub repo string, e.g. 'SparkPost/2web2ui'
+ * @returns {Object}
+ */
+function toQueryFriendly(githubRepo) {
+  const [owner, name] = githubRepo.split('/');
+
+  return {
+    owner: owner.toLowerCase(),
```
Comment (bkemper) [inline src/api.js:14]: nit: could call `toLowerCase` once before `split`.

File: package.json [hunk 5]
```diff
@@ -21,28 +24,35 @@
...
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^3.1.4",
     "eslint-plugin-promise": "^4.2.1",
     "eslint-plugin-standard": "^4.0.1",
     "husky": "^4.2.5",
+    "jest": "^26.6.3",
     "lint-staged": "^10.3.0",
     "prettier": "2.1.1"
   },
+  "devDependenciesComments": {
```
Comment (bkemper) [inline package.json:47]: 🥇

File: src/index.js [hunk 6]
```diff
@@ -1,68 +1,73 @@
...
-  const vulnerabilityAlerts = formatVulnerabilityAlerts(data)
+async function main() {
+  const { data } = await getSecurityVulnerabilities({
+    githubRepo: process.env.GITHUB_REPOSITORY,
+    githubToken: process.env.GITHUB_TOKEN,
+  });
+  const vulnerabilityAlerts = formatVulnerabilityAlerts(data);
 
-  if (!vulnerabilityAlerts) return console.log('No security vulnerabilities found.')
+  if (Boolean(vulnerabilityAlerts) && vulnerabilityAlerts.length === 0)
```
Comment (bkemper) [inline src/index.js:14]: I don't think you need this boolean check since you added a guard in the helper.

**Review by bkemper — APPROVED** [review]
👏

**Pull request merged**